src/ext/java/org/opentripplanner/ext/flex/FlexibleTransitLeg.java

```diff
@@ -59,27 +58,27 @@ public boolean isTransitLeg() {
 
   @Override
   public Agency getAgency() {
-    return edge.getTrip().getRoute().getAgency();
+    return getTrip().getRoute().getAgency();
   }
 
   @Override
   public Operator getOperator() {
-    return edge.getTrip().getOperator();
+    return getTrip().getOperator();
   }
 
   @Override
   public Route getRoute() {
-    return edge.getTrip().getRoute();
+    return getTrip().getRoute();
   }
 
   @Override
   public Trip getTrip() {
-    return edge.getTrip();
+    return edge.getFlexTrip().getTrip();
   }
 
   @Override
   public TraverseMode getMode() {
-    return TraverseMode.fromTransitMode(edge.getTrip().getRoute().getMode());
+    return TraverseMode.fromTransitMode(getTrip().getRoute().getMode());
```

Comment:
Is this correct? The `getTrip().getRoute().getMode()` get the route default mode, while the Trip may override it.

Reply:
That is true. I've changed it, even tough the error was already in the old code.